reliability-engineering/terraform/modules/concourse-worker-pool/autoscaling-group.tf

```diff
@@ -10,9 +10,24 @@ resource "aws_autoscaling_group" "concourse_worker" {
     "OldestInstance",
   ]
 
-  launch_template {
-    id      = aws_launch_template.concourse_worker.id
-    version = "$Latest"
+  mixed_instances_policy {
+    launch_template {
+      launch_template_specification {
+        launch_template_id = aws_launch_template.concourse_worker.id
+        version            = "$Latest"
+      }
+
+      dynamic "override" {
+        for_each = var.spot_instance_types
```

Comment:
is it ok if `spot_instance_types` is an empty list here? (thinking about defaults)

Would it make sense to have this as `coalescelist(var.spot_instance_types, [var.instance_type])` ?

The alternative being, we just add suitable instance types to all the existing worker groups

Reply:
Sure i'm happy to do this.

> The alternative being, we just add suitable instance types to all the existing worker groups

This is only necessary where we set `on_demand_percentage` lower than 100.  And in that case, we should probably explicitly set `spot_instance_types` anyway to maximise spot instance pools.